File: app/src/main/java/com/confessionsearch/release1/notesDBClassHelper.java
package com.confessionsearch.release1;

import android.content.Context;
import android.content.Intent;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;
import androidx.room.*;
import androidx.room.RoomDatabase;

@Database(entities = {Notes.class},version = 1)
public abstract class notesDBClassHelper extends RoomDatabase {
    public static final String DATABASE_NAME = "confessionsearchNotes_db";
    private static notesDBClassHelper instance;

    static notesDBClassHelper getInstance(final Context context) {
        if (instance == null) {
            instance = Room.databaseBuilder(context.getApplicationContext(), notesDBClassHelper.class, DATABASE_NAME).build();
            return instance;
        }
        return instance;
    }

    public abstract NoteDao getNoteDao();
}

File: app/src/main/java/com/confessionsearch/release1/databaseshelpers/notesDBClassHelper.kt
@file:Suppress("ClassName")

package com.confessionsearch.release1.databaseshelpers

import android.content.Context
import androidx.room.AutoMigration
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import com.confessionsearch.release1.data.notes.NoteDao
import com.confessionsearch.release1.data.notes.Notes

@Database(
    entities = [Notes::class],
    version = 3,
    exportSchema = true,
    autoMigrations = [AutoMigration(from = 1, to = 2), AutoMigration(from = 2, to = 3)]
)
abstract class notesDBClassHelper : RoomDatabase() {
    abstract val noteDao: NoteDao?

    companion object {
        private const val DATABASE_NAME = "confessionsearchNotes_db"
        private var instance: notesDBClassHelper? = null

        @JvmStatic
        fun getInstance(context: Context): notesDBClassHelper? {
            if (instance == null) {
                instance = Room.databaseBuilder(
                    context.applicationContext,
                    notesDBClassHelper::class.java,
                    DATABASE_NAME
                ).build()
                return instance
            }
            return instance
        }
    }
}